Tensile/SolutionSelectionLibrary.py

```diff
@@ -0,0 +1,123 @@
+################################################################################
+# Copyright (C) 2016-2019 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell cop-
+# ies of the Software, and to permit persons to whom the Software is furnished
+# to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IM-
+# PLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNE-
+# CTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+################################################################################
+
+from .Common import print1, print2, HR, printExit, defaultAnalysisParameters, globalParameters, pushWorkingPath, popWorkingPath, assignParameterWithDefault, startTime, ProgressBar, printWarning
+from .SolutionStructs import Solution
+from . import SolutionLibrary
+from . import Utils
+from . import YAMLIO
+from . import __version__
+
+from copy import deepcopy
+from sys import stdout
+import array
+import csv
+import os
+import time
+
+try:
+  import yaml
+except ImportError:
+  printExit("You must install PyYAML to use Tensile (to parse config files). See http://pyyaml.org/wiki/PyYAML for installation instructions.")
+
+def getSummationKeys(header):
+  keys=[]
+  for i in range(7, len(header)):
+    keystr = header[i].split("=")[1].strip()
+    key = int(keystr)
+    keys.append(key)
+  return keys
+
+def makeKey(row):
+  key=row[3]
+  for i in range(4, 7):
+    key += "_%s" % row[i].strip()
+  return key
+
+def analyzeSolutionSelection( problemType, problemSizeGroups):
+
+  dataFileNameList = []
+  performanceMap = {}
+  solutionData = {}
+  allSolutions = []
+
+  for problemSizeGroup in problemSizeGroups:
+    problemSizes = problemSizeGroup[0]
+    dataFileName = problemSizeGroup[3]
```

Comment:
We should make new sections of the YAML maps instead of arrays to avoid these magic numbers.